--- /dev/null
+++ ../library/Zend/Controller/Request/Http.php
@@ -1041,7 +1041,9 @@
      */
     public function getScheme()
     {
-        return ($this->getServer('HTTPS') == 'on') ? self::SCHEME_HTTPS : self::SCHEME_HTTP;
+        return (strtolower((string)$this->getServer('HTTPS')) == 'on') || $this->getServer('HTTP_X_FORWARDED_PROTO') == 'https' ?
+            self::SCHEME_HTTPS :
+            self::SCHEME_HTTP;
     }

     /**
