--- ../library/Zend/Auth/Adapter/Http/Resolver/File.php
+++ ../library/Zend/Auth/Adapter/Http/Resolver/File.php
@@ -157,7 +157,7 @@

         // No real validation is done on the contents of the password file. The
         // assumption is that we trust the administrators to keep it secure.
-        while (($line = fgetcsv($fp, 512, ':')) !== false) {
+        while (($line = fgetcsv($fp, 512, ':', '"', '\\')) !== false) {
             if ($line[0] == $username && $line[1] == $realm) {
                 $password = $line[2];
                 fclose($fp);
